synapse/storage/databases/main/events.py

```diff
@@ -1770,10 +1770,12 @@ def _handle_insertion_event(self, txn: LoggingTransaction, event: EventBase):
             # Not a insertion event
             return
 
-        # Skip processing a insertion event if the room version doesn't
-        # support it.
+        # Skip processing an insertion event if the room version doesn't
+        # support it or the event is not from the room creator.
         room_version = self.store.get_room_version_txn(txn, event.room_id)
-        if not room_version.msc2716_historical:
+        create_event = self.store.get_create_event_for_room_txn(txn, event.room_id)
```

Comment:
`get_create_event_for_room` already exists. This PR adds `get_create_event_for_room_txn`

As an alternative way to look this up, the [`rooms` table also has the `creator` field](https://github.com/matrix-org/synapse/blob/7afb615839a2df05d39f87718016d278ebdadf5c/synapse/storage/schema/main/full_schemas/54/full.sql.postgres#L941).

Here are couple places in the code where we try to get the `creator`:

 - [`synapse/event_auth.py#L263-L267`](https://github.com/matrix-org/synapse/blob/7afb615839a2df05d39f87718016d278ebdadf5c/synapse/event_auth.py#L263-L267)
  - [`synapse/storage/databases/main/events.py#L1109-L1114`](https://github.com/matrix-org/synapse/blob/7afb615839a2df05d39f87718016d278ebdadf5c/synapse/storage/databases/main/events.py#L1109-L1114)